platforms/software/maven/src/integTest/groovy/org/gradle/api/publish/maven/MavenPublishSnapshotIntegTest.groovy

```diff
@@ -195,4 +195,46 @@ class MavenPublishSnapshotIntegTest extends AbstractMavenPublishIntegTest {
             snapshotVersions == ["1.0-SNAPSHOT"]
         }
     }
+
+    def "published metadata after artifacts"() {
+        settingsFile << 'rootProject.name = "snapshotPublish"'
+        buildFile << """
+    apply plugin: 'java'
+    apply plugin: 'maven-publish'
+
+    group = 'org.gradle'
+    version = '1.0-SNAPSHOT'
+
+    publishing {
+        repositories {
+            maven { url "${mavenRepo.uri}" }
+        }
+        publications {
+            pub(MavenPublication) {
+                from components.java
+            }
+        }
+    }
+"""
+        def module = mavenRepo.module('org.gradle', 'snapshotPublish', '1.0-SNAPSHOT')
+
+        when:
+        succeeds 'publish', '-i'
+
+        then:
+        def initialVersion = module.publishArtifactVersion
+        def publishTaskOutputLines = result.getGroupedOutput().task(':publishPubPublicationToMavenRepository').getOutput().split("\\R")
+        def expectedPublishedLog = ["Uploading snapshotPublish-${initialVersion}.jar", "Uploading snapshotPublish-${initialVersion}.pom", "Uploading snapshotPublish-${initialVersion}.module", 'Uploading maven-metadata.xml', 'Uploading maven-metadata.xml']
```

Comment:
💅 Isn't there a better way to get that information? E.g. requesting the maven repo for the artifact upload time.

The current way looks fragile to me - logs format may change, and the publishing can be potentially parallelized (meaning races on stdout)

Reply:
This test only use a file based repo, so not immediately.

Changes to log format or to parallelization would impact this test. While the former would be noise, it would also not be hard to update. The later would indicate that care must be taken.